src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -1328,7 +1329,8 @@ internal void ParseNestedProjects()
                     break;
                 }
 
-                if (String.IsNullOrWhiteSpace(str))
+                // Ignore empty line or comment
+                if (String.IsNullOrWhiteSpace(str) || str[0] == CommentStartChar)
```

Comment:
The `#` is not necessarily is the first character of the string, it's the first `non-whitespace` character.

A `TrimStart()` would be needed, preferably on a `ReadOnlySpan<char>` view of the string to avoid allocations.

Reply:
In all affected functions is line got via `ReadLine()` method (https://github.com/dotnet/msbuild/blob/301307bc82c040ad758da0587b5a2929015f726c/src/Build/Construction/Solution/SolutionFile.cs#L438). This method trims content of the line. So my application should be still valid.

Reply:
Sorry! You are right, it is correct, I completely overlooked this method earlier, even when double checking.